File: 251002/Troitsky/Lab1-5/Lab1.Application/Services/PostService.cs
// Было нужно для 1-3 лабораторных

/*
using Lab1.Core.Abstractions;
using Lab1.Core.Models;
using Lab1.Core.Contracts;
using System.Numerics;
using FluentValidation;
using Lab1.Application.Exceptions;
namespace Lab1.Application.Services
{
    public class PostService : IPostService
    {
        private IPostRepository _repository;
        private IValidator<Post> _validator;
        public PostService(IPostRepository repository, IValidator<Post> validator) => (_repository, _validator) = (repository, validator);
        public PostResponseTo CreatePost(Post post)
        {
            var res = _validator.Validate(post);
            if (!res.IsValid)
                throw new IncorrectDataException("invalid input");
            try
            {
                return _repository.Create(post);
            }
            catch
            {
                throw new IncorrectDatabaseException("DB error");
            }
        }
        public bool DeletePost(ulong id)
        {
            try
            {
                return _repository.Delete(id);
            }
            catch
            {
                throw new IncorrectDatabaseException("DB error");
            }
        }
        public PostResponseTo? GetPost(ulong id)
        {
            try
            {
                return _repository.Get(id);
            }
            catch
            {
                throw new IncorrectDatabaseException("DB error");
            }
        }
        public List<PostResponseTo> GetAllPosts()
        {
            try
            {
                return _repository.GetAll();
            }
            catch
            {
                throw new IncorrectDatabaseException("DB error");
            }
        }
        public PostResponseTo? UpdatePost(Post post, ulong id)
        {
            var res = _validator.Validate(post);
            if (!res.IsValid)
                throw new IncorrectDataException("invalid input");
            try
            {
                return _repository.Update(post, id);
            }
            catch
            {
                throw new IncorrectDatabaseException("DB error");
            }
        }
    }
}
*/

File: 251001/Karpekov/lab1/TaskRest/src/main/java/lab1/services/PostService.java
package lab1.services;

import lab1.datalayer.DataLayer;
import lab1.datalayer.FieldChecker;
import lab1.datalayer.InMemoryDataLayer;
import lab1.models.Post;
import org.springframework.stereotype.Component;

import java.util.ArrayList;

@Component
public class PostService {

    private final DataLayer dataLayer;
    public PostService(){
        dataLayer = new InMemoryDataLayer();
    }

    public ArrayList<Post> GetPosts(){
       return dataLayer.GetPosts();
    }

    public Post GetPostById(long Id) throws IllegalArgumentException{
        Post res = dataLayer.GetPost(Id);
        return res;
    }

    public boolean CreatePost(Post post){
        if (FieldChecker.CheckPostFields(post)){
            dataLayer.CreatePost(post);
            return post.getId() != -1;
        }else
            return false;
    }

    public Post DeletePost(long Id){
        return dataLayer.DeletePost(Id);
    }

    public Post UpdatePost(Post post){
        if (FieldChecker.CheckPostFields(post))
            return dataLayer.UpdatePost(post);
        else
            return null;
    }

    public DataLayer getDataLayer(){
        return dataLayer;
    }

}
